### PR TITLE
Stage2: Initial behavior tests succeeding

### PR DESCRIPTION
**Pull request by Luukdegram**
With this, we get the initial few behavior tests succeeding for the wasm backend.

Note: During the implementation of this, I came across 2 memory leaks in stage2. The fixes for those are both included in this PR.

### CODE REVIEW

File: src/Module.zig [hunk 1]
```diff
@@ -1226,19 +1226,20 @@ pub const Fn = struct {
...
         if (ret_ty.tag() == .generic_poison) {
             return null;
         }
         if (ret_ty.zigTypeTag() == .ErrorUnion) {
             if (ret_ty.errorUnionSet().castTag(.error_set_inferred)) |payload| {
-                return &payload.data.map;
+                return &payload.data;
             }
         }
         return null;
```
Comment (Luukdegram) [inline src/Module.zig:1245]: @andrewrk Let me know if you're okay with those changes to fix the memory leak

File: src/Module.zig [hunk 2]
```diff
@@ -1301,6 +1302,7 @@ pub const Namespace = struct {
             key.destroy(mod);
         }
         anon_decls.deinit(gpa);
+        ns.usingnamespace_set.deinit(gpa);
```
Comment (Luukdegram) [inline src/Module.zig:1305]: Same as above (to fix a memory leak)

File: src/Module.zig [hunk 3]
```diff
@@ -1332,6 +1334,8 @@ pub const Namespace = struct {
             child_decl.destroy(mod);
         }
         anon_decls.deinit(gpa);
+
+        ns.usingnamespace_set.deinit(gpa);
```
Comment (Luukdegram) [inline src/Module.zig:1338]: Same as bove

**Review by Luukdegram — COMMENTED** [review]
Implemented some fixes for the memory leaks, but as my knowledge of this part of the compiler is quite limited, I'd like to make sure this is being reviewed.

**Review by andrewrk — APPROVED** [review]
Both memory leak fixes look great! Sorry for the trouble and thank you for fixing them.

**Pull request merged**